spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationProperties.java

```diff
@@ -295,4 +302,86 @@ public void setMessageMappingEnabled(boolean messageMappingEnabled) {
 
 	}
 
+	public static class Poller {
+
+		/**
+		 * Maximum of messages to poll per polling cycle.
+		 */
+		private Long maxMessagesPerPoll;
+
+		/**
+		 * How long to wait for messages on poll.
+		 */
+		private Duration receiveTimeout;
```

Comment:
This field could have a default value that matches `PollerMetadata.DEFAULT_RECEIVE_TIMEOUT`. That would then make the default available in auto-complete when configuring the property.

Reply:
Thank you Andy for feedback!
If I recall correctly, you or Stéphane pointed us in the past that it is better to keep configuration properties beans free from target libraries dependencies, so I decided to go with that sensitive defaults which comes with a `PollerMetadata` instance and what could be changed in the past if we would use values over here instead of those constant references. If that is still OK, I can change it as you suggest to point to those `PollerMetadata` constants. 
Or I can use their values directly for time being. That's in the case if we don't go with nulls as I have done now.

Reply:
We typically duplicate the default without referencing anything in the library. We can then test that our default matches the library's default, as you have already done in this case.